### PR TITLE
Add FuchsiaReady for handling custom signals

### PR DESCRIPTION
**Pull request by cramertj**
cc @raggi

### CODE REVIEW

File: src/sys/fuchsia/ready.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
+use event_imp::{Ready, ready_as_usize, ready_from_usize};
+pub use magenta_sys::mx_signals_t;
+use std::ops;
+
+// The following impls are valid because Fuchsia and mio both represent
+// "readable" as `1 << 0` and "writable" as `1 << 2`.
```
Comment (alexcrichton) [inline src/sys/fuchsia/ready.rs]: Could this be a debug assert somewhere?

File: src/sys/fuchsia/ready.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+    #[inline]
+    pub fn has_any(&self, signals: mx_signals_t) -> bool {
+        !(Ready::from(signals) & self.0).is_empty()
+    }
+
+    /// Returns the `FuchsiaReady` as raw magenta signals.
+    /// This function is just a more explicit, non-generic version of
+    /// `FuchsiaReady::into`.
+    #[inline]
+    pub fn into_raw_signals(self) -> mx_signals_t {
```
Comment (alexcrichton) [inline src/sys/fuchsia/ready.rs]: Perhaps `into_mx_signals` as a name?

File: src/sys/fuchsia/ready.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    /// magenta signals.
+    #[inline]
+    pub fn has_all(&self, signals: mx_signals_t) -> bool {
+        (Ready::from(signals) - self.0).is_empty()
+    }
+
+    /// Returns whether or not the `FuchsiaReady` contains any of the specified
+    /// magenta signals.
+    #[inline]
+    pub fn has_any(&self, signals: mx_signals_t) -> bool {
```
Comment (alexcrichton) [inline src/sys/fuchsia/ready.rs]: Could this and `has_all` be removed in favor of bit twiddling? (just thinking for consistency between Unix and Fuchsia)

File: src/sys/fuchsia/ready.rs (hunk 2 above)
Comment (cramertj) [inline src/sys/fuchsia/ready.rs]: Sure!

File: src/sys/fuchsia/ready.rs (hunk 3 above)
Comment (cramertj) [inline src/sys/fuchsia/ready.rs]: Sure-- I'll implement some more of the boolean operators for these.

File: src/sys/fuchsia/ready.rs (hunk 1 above)
Comment (cramertj) [inline src/sys/fuchsia/ready.rs]: Sure! I don't think it's going to change anytime soon, but better to be safe than wind up with untraceable errors.

**Comment by alexcrichton** [comment]
👍

**Pull request merged**